### ISSUE TITLE
flatMap not detected as part of a lodash chain

### ISSUE THREAD

**Issue by lukeapage** [wix/eslint-plugin-lodash#49]
this
```
_.chain(a).flatMap('b').map('c').value()
```
is hitting prefer-lodash-method because `flatMap` isn't marked chainable.

But am also confused by the lodash documentation - I raised this..
https://github.com/lodash/lodash/issues/2166
because it looks like everything is chainable.. so am not sure whether everything should be chainable or just to add `flatMap` to the chainable list..

**Comment by ganimomer** [comment]
The lodash docs are a bit fuzzy on the matter (there's separate docs for [`_.prototype.flatMap`](https://lodash.com/docs#prototype-flatMap)), but it looks like you're right, this is a bug in the plugin.
I tried running `_(arr).flatMap()` and it returned a wrapper, so it's chainable.

**Comment by lukeapage** [comment]
Even things the docs say arent possible e.g. join return a wrapper. We will see what jdalton says in a bit anyway..

**Comment by ganimomer** [comment]
I answered you by myself in the issue. The docs **are** unclear about `_.flatMap`, and what you reported here looks like an actual bug, so both issues you opened are valid. :smile:

**Comment by lukeapage** [comment]
So now I understand - it looks to me like there is another bug - the lodash utils exits early if the method is not chainable, without checking to see if its implicit or explicit...

**Comment by lukeapage** [comment]
here: https://github.com/wix/eslint-plugin-lodash/blob/master/src/util/lodashUtil.js#L94

**Comment by ganimomer** [comment]
Right again! I'll open a separate issue.
